meson.build

```diff
@@ -1713,7 +1713,12 @@ public_programs = []
 dbus_programs = []
 dbus_interfaces_dir = get_option('dbus-interfaces-dir')
 if dbus_interfaces_dir == ''
-   dbus_interfaces_dir = get_option('datadir') + '/dbus-1'
+   if meson.is_cross_build()
+      warning('D-Bus Introspection XML export not supported for cross-compilation.')
```

Comment:
If it is not supported, then refuse all 'dbus-interface-dir' except for 'no' and ''? How about the following?
```
dbus_interfaces_dir = get_option('dbus-interfaces-dir')
if meson.is_cross_build()
        if dbus_interface_dir != '' and dbus_interface_dir != 'no'
                error('D-Bus Introspection XML export not supported for cross-compilation.')
        dbus_interface_dir = 'no'
elif dbus_interface_dir = ''
        dbus_interfaces_dir = get_option('datadir') + '/dbus-1'
endif
```

Reply:
BTW, maybe `get_option('datadir')` -> `datadir`??

Reply:
> If it is not supported, then refuse all values except for 'no' and ''?

The thing is it might be supported if for example there is qemu installed or x86 on x86_64 system. There is a [`meson.can_run_host_binaries`](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesoncan_run_host_binaries) to figure out if it can run so maybe using this instead of `is_cross_build`